config/crd/patches/fix-tekton-params.yaml

```diff
@@ -1,6 +1,16 @@
 # replace the generated schema for params[].value with a custom one
 # (Tekton uses custom marshalling and unmarshalling for param values)
 - op: replace
-  path: /spec/versions/0/schema/openAPIV3Schema/properties/spec/properties/selectors/items/properties/pipelineRef/properties/params/items/properties/value
+  path: /spec/versions/0/schema/openAPIV3Schema/properties/spec/properties/selectors/items/properties/pipelineRef/properties/params/items
   value:
-    type: string
+    description: |
+      Tekton parameter definition, see https://tekton.dev/docs/pipelines/pipelineruns/#specifying-parameters
+      {
+        name: string,
+        value: string | [string] | {string: string}
+      }
+    type: object
+    # it is not possible to define a structural schema that would allow all the valid values
+    # define params as opaque objects, don't prune any fields
+    # https://kubernetes.io/blog/2019/06/20/crd-structural-schema/#extensions
+    x-kubernetes-preserve-unknown-fields: true
```

Comment:
This whole commit can be dropped if we want to allow only string values for pipelineRef params